### PR TITLE
Feature/fetch data

### PR DESCRIPTION
**Pull request by PaigeVannelli**
## Pull Request
<br>
### Description
* Is this a feature or a fix?
 - Feature: we added our fetch requests to get all movie posters to display on main page. We also added a fetch request to our MovieDetails component to fetch a particular movies data when a poster is clicked on. We then changed the detailed display to include all of the fetched movie details. For both fetch requests we added error handling to handle both loading and 500 errors. 
 - We also cleaned up some of the formatting to make displaying the fetched data cleaner
 
* Where should the reviewer start?
 - Review App componentDidMount fetch request and review the MovieDetails component 
 - Review app styles 
 
* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
 - This now allows the app to get and display fetched data from the servers. It also allows the users to see error messages based on what kind of error is happening.  
 
* What outside features or research did you use to implement this fix?
 - n/a
* How should this be tested?
 - open the website, click on posters to see detailed movie data. Break the fetch url to see error message
<br>
***
#### Please review considerations below:
- [X] Follows Turing Style Guidelines
- [X] Changes generate no new warnings or errors
- [X] Checked code for and corrected any spelling errors
- [ ] README has been updated(if applicable)

### CODE REVIEW

**Comment by PaigeVannelli** [comment]
@letakeane Here's our PR!

File: src/App.js [hunk 1]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
   constructor() {
     super();
     this.state = {
-      movieData: movieData,
-      view: 'mainPage'
+      // movieData: movieData,
```
Comment (letakeane) [inline src/App.js:12]: You've probably already removed this commented out code :)

File: src/App.js [hunk 2]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
+      movieData: {
```
Comment (letakeane) [inline src/App.js:13]: I'm curious about this nested bit of state - what's the advantage of this over just storing an array of movies in state directly?

File: src/App.js [hunk 3]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
+      currentMovieId: 0,
+      error: '',
     }
   }
 
-goToMain = () => {
-  this.setState({view: 'mainPage'})
-}
+  componentDidMount() {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (letakeane) [inline src/App.js:23]: If you haven't already (sorry for the lateness of this PR), be sure to move your fetches into their own file.

File: src/App.js [hunk 4]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
+      return <h1 className='error'>Failed to load</h1>
+    }
+  }
+
+  goToMain = () => {
+    this.setState({view: 'mainPage', currentMovieId: 0})
+  }
+
+  displayMovieDetails = (id) => {
+    console.log(id)
```
Comment (letakeane) [inline src/App.js:46]: Kill this console log eventually

File: src/App.js [hunk 5]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
+    this.setState({view: 'detailedView', currentMovieId: id})
+  }
 
   render() {
     return (
       <main className='main-page'>
         <nav className='nav'>
-          <button className='main-logo' onClick={this.goToMain}><img src={logo} className='movie-reel-logo'/>Rancid<br>
-          </br>Tomatillos</button>
+          <button className='main-logo' onClick={this.goToMain}><img src={logo} className='movie-reel-logo'/>Cinematic</button>
```
Comment (letakeane) [inline src/App.js:54]: Don't forget an `alt` tag :D

File: src/App.js [hunk 6]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
         </nav>
+        {this.checkIfLoading()}
```
Comment (letakeane) [inline src/App.js:56]: Love that you moved this logic into a helper method

File: src/App.js [hunk 7]
```diff
@@ -9,28 +9,54 @@ class App extends Component {
...
+        {this.handleIfFailed()}
         {this.state.view === 'mainPage' && <Movies movieData={this.state.movieData} displayMovieDetails={this.displayMovieDetails}/>}
```
Comment (letakeane) [inline src/App.js:58]: It's generally convention to put the component being rendered on its own line

File: src/MovieDetails/MovieDetails.js [hunk 8]
```diff
@@ -2,29 +2,79 @@ import React, { Component } from 'react'
...
-                </div>
-                <p className='summary'>
-                    sdkasdnkask ldankk sdlnkl nask lna snkdkandkna kdaka sasdkas 
-                </p>
-            </section>
+            <div className='movies-div'>
+                {this.checkIfLoading()}
+                {this.handleIfFailed()}
+                {(!this.state.error && this.state.title) &&
+                    <section style={{background:
```
Comment (letakeane) [inline src/MovieDetails/MovieDetails.js:57]: Ooooh I'm not a fan of this inline styling. Can you tie this styling to a class instead??

File: src/Movies/Movies.js [hunk 9]
```diff
@@ -3,9 +3,10 @@ import Poster from '../Poster/Poster.js'
 import './Movies.css'
 
 const Movies = (props) => {
+    // console.log(props)
```
Comment (letakeane) [inline src/Movies/Movies.js:6]: Remember, when checking props or state, it's best to do this not with a console log but by using the React Dev Tools

File: src/Movies/Movies.js [hunk 10]
```diff
@@ -3,9 +3,10 @@ import Poster from '../Poster/Poster.js'
...
     const allMovies = props.movieData.movies.map(movie => {
         return (
-            <Poster image={movie.poster_path} key={movie.id} displayMovieDetails={props.displayMovieDetails}/>
+            <Poster image={movie.poster_path} key={movie.id} id={movie.id} displayMovieDetails={props.displayMovieDetails}/>
```
Comment (letakeane) [inline src/Movies/Movies.js:9]: If you're storing the ID for later use, it's better to use a data-id attribute instead of just the id attribute

**Review by letakeane — COMMENTED** [review]
Apologies for the lateness of the review! Ignore whatever is no longer relevant.

**Pull request merged**